addons/mrp_account/models/mrp_production.py

```diff
@@ -34,13 +34,23 @@ def _compute_analytic_account_id(self):
             self.analytic_account_id = self.bom_id.analytic_account_id
 
     def write(self, vals):
+        origin_analytic_account = self.analytic_account_id
         res = super().write(vals)
         if vals.get('name'):
             for production in self:
                 production.move_raw_ids.analytic_account_line_id.ref = production.display_name
                 for workorder in production.workorder_ids:
                     workorder.mo_analytic_account_line_id.ref = production.display_name
                     workorder.mo_analytic_account_line_id.name = _("[WC] %s", workorder.display_name)
+        if 'analytic_account_id' in vals and self.state == 'done':
```

Comment:
As discussed, we should update the line for all stage except `draft` and `cancel`.

Reply:
I agree that the account analytic can be modified regardless of the MO stage, but the account analytic lines are created only when the MO is done, so it will be updated or created only when the status is done

Reply:
@DjamelTouati the analytical lines are posted as soon as there is any consumption of raw materials or work center usage, so it should be updated at all times, except for the draft stage. cc @yhu-odoo

Reply:
Ok perfect, I updated the PR, thanks for the explanation!